TheWarWithin/RogueAssassination.lua

```diff
@@ -2907,61 +2916,78 @@ spec:RegisterStateExpr( "envenom_at_5cp", function ()
     return settings.envenom_pool_5_points and 1 or 0
 end )--]]
 
+spec:RegisterSetting( "fok_critical_cp_prediction", "predict", {
+    name = strformat( "%s Critical Combo Point Prediction", Hekili:GetSpellLinkWithTexture( 51723 ) ),  -- Fan of Knives
+    desc = strformat( "%sThis setting controls how %s predicts critical strikes for combo point generation.%s " ..
+                      "|n|n%sThis option only works if the %s talent is selected.%s\n\n" ..
+                      "|cFF00FF00• Predict:|r Assumes a certain number of critical hits will occur based on your current critical strike chance and number of nearby enemies.\n\n" ..
+                      "|cFF00FF00• Predict Conservatively:|r The predicted number of combo points is reduced by 1 in order to prevent more jarring rotation changes from unlucky critical strikes.\n\n" ..
+                      "|cFF00FF00• Do Not Predict:|r Disables critical prediction, using only guaranteed combo points from casting %s.",
+                      "|cFFFFD100", Hekili:GetSpellLinkWithTexture( 51723 ), "|r",
+                      "|cFFFF0000", Hekili:GetSpellLinkWithTexture( 14190 ), "|r",
+                      Hekili:GetSpellLinkWithTexture( 51723 )
+    ),
+    type = "select",
+    values = {
+        ["predict"] = "Predict",
+        ["predict_conservatively"] = "Predict Conservatively",
+        ["do_not_predict"] = "Do Not Predict"
+    },
+    width = 1.5,
+} )
+
 spec:RegisterSetting( "envenom_pool_pct", 0, {
-    name = "Energy % for |T132287:0|t Envenom",
-    desc = "If set above 0, |T132287:0|t Envenom will ONLY be used when you have at least this percentage of your Energy.",
+    name = strformat( "Minimum Energy %% to Cast %s", Hekili:GetSpellLinkWithTexture( 32645 ) ),
```

Comment:
~~`to Cast`~~ -> `for`

Probably need to validate that this is true in the current priority.

Reply:
Any pooling appears to already be handled by the APL pooling variable, not actually on envenom lines. Nothing in any written guides suggest pooling specifically for envenom either.

Reply:
We could make that `pool_resource` line make use of the setting, or we can remove the setting.

```
actions.direct+=/variable,name=extra_pool,condition=(40+30*talent.hand_of_fate.rank-15*talent.vicious_venoms.rank)>settings.envenom_pool_pct,value=energy.max*(0.4+0.3*talent.hand_of_fate.rank-0.15*talent.vicious_venoms.rank)-action.envenom.cost,value_else=energy.max*(settings.envenom_pool_pct/100)-action.envenom.cost
actions.direct+=/pool_resource,for_next=1,extra_amount=variable.extra_pool,if=active_dot.envenom=0
```

This would make the setting allow you to force a higher pooling threshold, but still abide by the talent-based threshold set in the priority?

(It might be best to just remove this and see what happens.)